Source/RxSwift/and.swift

```diff
@@ -2,18 +2,101 @@
 //  and.swift
 //  RxSwiftExt
 //
-//  Created by Joan Disho on 19.10.17.
+//  Created by Florent Pillet on 26/11/17
 //  Copyright © 2017 RxSwiftCommunity. All rights reserved.
 //
+import Foundation
+import RxSwift
 
-extension Observable where E == Bool {
+extension ObservableType where E == Bool {
+	/**
+	Converts a source sequence of Bool values to a single final result `true` or `false`
+	unless no value is received, in which case the resulting `Maybe` sequence will just complete.
 
-    func and<C: Collection>(_ collection: C) -> Observable<Bool> where C.Iterator.Element: ObservableType, C.Iterator.Element.E == E {
-        return Observable.combineLatest(collection).map { $0.reduce(true, { $0 && $1 }) }
-    }
-    
-    func and<O: ObservableType>(_ sources: O ...) -> Observable<Bool> where O.E == Bool {
-        return self.and(sources)
-    }
-    
+	If a `false` value is emitted, the resulting sequence immediately completes with a `false` result.
+	If only `true` values are emitted, the resulting sequence completes with a `true` result once the
+	source sequence completes.
+	If no value is emitted, the resulting sequence completes with no value once the source sequence completes.
+
+	Use `asSingle()` or `asObservable()` to convert to your requirements.
+	*/
+	public func and() -> Maybe<E> {
+		return Maybe.create { observer in
+			var gotValue = false
+			return self.subscribe { event in
+				switch event {
+				case .next(let value):
+					if !value {
```

Comment:
Not really because I forgot an `else` to avoid setting `gotValue` when not needed.

Reply:
Yeah that was my point though, a guard would not allow fall-through anyways so `gotValue` wouldn't get set unless needed. And since this feels like an "early return" scenario, guard (to me at least) makes more sense. Either way works for me :)